infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/GeocodingService.kt

```diff
@@ -102,32 +105,43 @@ class GeocodingService(
         trackNumberId: DomainId<TrackLayoutTrackNumber>?,
     ) = if (trackNumberId is IntId) getGeocodingContext(publicationState, trackNumberId) else null
 
+    fun getGeocodingContext(geocodingContextCacheKey: GeocodingContextCacheKey) =
+        geocodingCacheService.getGeocodingContext(geocodingContextCacheKey)
+
     fun getGeocodingContext(
         publicationState: PublishType,
         trackNumberId: IntId<TrackLayoutTrackNumber>,
     ): GeocodingContext? =
-        geocodingDao.getGeocodingContextCacheKey(publicationState, trackNumberId)?.let(::getGeocodingContext)
+        geocodingDao.getLayoutGeocodingContextCacheKey(publicationState, trackNumberId)?.let(geocodingCacheService::getGeocodingContext)
```

Comment:
Ihan huomiona, näissä tämä vastuualuelogiikka menee ainakin kutsujen nimien osalta vähän jännästi/epäintuitiivisesti. Meillä on nyt `geocodingDao`, jolta ei haeta geokoodauskontekstia vaan kakkuavain, ja `geocodingCacheService`, jolta me ei haeta mitään ainakaan nimen perusteella kakkuun liittyvää vaan itse geokoodauskonteksti. Itse olettaisin näiden käyttäjinä noiden tulevan toisin päin.